charms/jupyter-ui/tests/unit/test_operator.py

```diff
@@ -142,3 +171,77 @@ def test_deploy_k8s_resources_success(
         harness.charm._deploy_k8s_resources()
         k8s_resource_handler.apply.assert_called()
         assert isinstance(harness.charm.model.unit.status, MaintenanceStatus)
+
+    @pytest.mark.parametrize(
+        "config_key,expected_images",
+        [
+            ("jupyter-images", ["jupyterimage1", "jupyterimage2"]),
+            ("vscode-images", ["vscodeimage1", "vscodeimage2"]),
+            ("rstudio-images", ["rstudioimage1", "rstudioimage2"]),
+        ],
+    )
+    @patch("charm.KubernetesServicePatch", lambda x, y, service_name: None)
+    @patch("charm.JupyterUI.k8s_resource_handler")
+    def test_notebook_selector_images_config(
+        self, k8s_resource_handler: MagicMock, harness: Harness, config_key, expected_images
+    ):
+        """Test that updating the images config works as expected for:
+        Jupyter images, VSCode images, and RStudio images ."""
+        expected_images_yaml = yaml.dump(expected_images)
```

Comment:
minor suggestion - you can ignore if you dont like it.  

To me, tests are usually most readable if formatted in the [arrange-act-assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) pattern.  you basically do this already, but to me its helpful to have the comments

like:
```
# Arrange
expeted_images_yaml = ...
harness ...
...

# Act
actual_images = harness.charm...

# Assert
assert actual_images ...
```

Don't follow it religiously (sometimes you should break the pattern, like sometimes the Act/Assert really is the same thing), but often for me it helps remind me when I'm making a test do too many things

Reply:
good suggestion, followed it in unit tests